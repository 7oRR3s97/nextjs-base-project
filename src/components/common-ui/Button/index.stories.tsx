import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import Button from './index';

export default {
  title: 'CommonUI/Button',
  component: Button,
  argTypes: { onClick: { action: 'clicked' } },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Link = Template.bind({});
Link.args = {
  layout: 'link',
  children: 'button',
  fullWidth: false,
  isLoading: false,
  disabled: false,
  newWindow: false,
  href: undefined,
  icon: undefined,
  variantText: 'body1',
};

export const SelectedUnderlined = Template.bind({});
SelectedUnderlined.args = {
  ...Link.args,
  layout: 'selectedUnderlined',
};

export const Halfline = Template.bind({});
Halfline.args = {
  ...Link.args,
  layout: 'halfline',
};

export const Outlined = Template.bind({});
Outlined.args = {
  ...Link.args,
};

export const Outlinedblack = Template.bind({});
Outlinedblack.args = {
  ...Link.args,
  layout: 'outlinedBlack',
};

export const Icon = Template.bind({});
Icon.args = {
  ...Link.args,
  layout: 'icon',
};

export const IconUnderlined = Template.bind({});
IconUnderlined.args = {
  ...Link.args,
  layout: 'iconUnderlined',
};

export const Box = Template.bind({});
Box.args = {
  ...Link.args,
  layout: 'box',
};

export const Filled = Template.bind({});
Filled.args = {
  ...Link.args,
  layout: 'filled',
};

export const FilledBlack = Template.bind({});
FilledBlack.args = {
  ...Link.args,
  layout: 'filledBlack',
};

export const FilledWhite = Template.bind({});
FilledWhite.args = {
  ...Link.args,
  layout: 'filledWhite',
};

export const FilledBoxShadowMain = Template.bind({});
FilledBoxShadowMain.args = {
  ...Link.args,
  layout: 'filledBoxShadowMain',
};

export const FilledBoxShadowWhite = Template.bind({});
FilledBoxShadowWhite.args = {
  ...Link.args,
  layout: 'filledBoxShadowWhite',
};

export const Gradient = Template.bind({});
Gradient.args = {
  ...Link.args,
  layout: 'gradient',
};

export const Footer = Template.bind({});
Footer.args = {
  ...Link.args,
  layout: 'footer',
};

export const Quiz = Template.bind({});
Quiz.args = {
  ...Link.args,
  layout: 'quiz',
};

export const Text = Template.bind({});
Text.args = {
  ...Link.args,
  layout: 'text',
};

export const Gray = Template.bind({});
Gray.args = {
  ...Link.args,
  layout: 'gray',
};

export const GrayOutlined = Template.bind({});
GrayOutlined.args = {
  ...Link.args,
  layout: 'grayOutlined',
};

export const SquareUpDown = Template.bind({});
SquareUpDown.args = {
  ...Link.args,
  layout: 'squareUpDown',
};

export const Alert = Template.bind({});
Alert.args = {
  ...Link.args,
  layout: 'alert',
};
