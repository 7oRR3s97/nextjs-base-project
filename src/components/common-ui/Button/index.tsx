import React from 'react';
import Link from 'next/link';
import { ThemeProvider } from '@mui/material/styles';

import MuiButton, { ButtonProps } from '@mui/material/Button';
import CircularProgress from '@mui/material/CircularProgress';
import Typography from '@mui/material/Typography';

import { Variant } from '@mui/material/styles/createTypography';

import theme, { LoaderContainer, IconContainer } from './styles';

export type Layout =
  | 'link'
  | 'underlined'
  | 'selectedUnderlined'
  | 'halfline'
  | 'outlined'
  | 'outlinedBlack'
  | 'icon'
  | 'iconUnderlined'
  | 'box'
  | 'filled'
  | 'filledBlack'
  | 'filledWhite'
  | 'filledBoxShadowMain'
  | 'filledBoxShadowWhite'
  | 'gradient'
  | 'footer'
  | 'quiz'
  | 'text'
  | 'gray'
  | 'grayOutlined'
  | 'squareUpDown'
  | 'alert';

export interface Props extends ButtonProps {
  href?: string;
  newWindow?: boolean;
  isLoading?: boolean;
  layout?: Layout;
  variantText?: Variant;
  icon?: JSX.Element;
  fullWidth?: boolean;
}

const Button: React.FC<Props> = ({
  children,
  href,
  newWindow,
  isLoading,
  layout = 'link',
  disabled,
  className,
  variantText = 'body1',
  icon,
  fullWidth,
  ...props
}) => {
  const loader = React.useMemo(() => {
    if (!isLoading) return null;
    return (
      <LoaderContainer>
        <CircularProgress
          color="inherit"
          size={theme.typography[variantText].fontSize}
        />
      </LoaderContainer>
    );
  }, [isLoading, variantText]);

  const iconComponent = React.useMemo(() => {
    if (!icon) return null;
    return <IconContainer>{icon}</IconContainer>;
  }, [icon]);

  const buttonContent = React.useMemo(() => {
    return (
      <MuiButton
        variant={layout}
        disabled={disabled || isLoading}
        fullWidth={fullWidth}
        {...props}
      >
        {loader}
        {iconComponent}
        <Typography
          variant={variantText}
          sx={{
            fontFamily: 'inherit',
            fontWeight: 'inherit',
          }}
        >
          {children}
        </Typography>
      </MuiButton>
    );
  }, [
    children,
    disabled,
    isLoading,
    layout,
    fullWidth,
    loader,
    props,
    variantText,
    iconComponent,
  ]);

  const button = React.useMemo(() => {
    if (href) {
      if (newWindow) {
        return (
          <a
            target="_blank"
            rel="noreferrer"
            style={{ textDecoration: 'none' }}
            href={href}
          >
            {buttonContent}
          </a>
        );
      }
      return <Link href={href}>{buttonContent}</Link>;
    }
    return buttonContent;
  }, [buttonContent, href, newWindow]);

  return <ThemeProvider theme={theme}>{button}</ThemeProvider>;
};

export default Button;
