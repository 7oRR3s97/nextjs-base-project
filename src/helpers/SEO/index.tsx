import { useTranslation } from 'next-i18next';

export interface Metadata {
  title: string;
  image: string;
  description: string;
  ogType: string;
  url: string;
  page: string;
  locale: string;
  params?: { [key: string]: string | string[] };
}

// TODO: Alterar dados para SEO
const useSeo = (props: Partial<Metadata>) => {
  const { t: userPageT } = useTranslation('seo-user-page');
  const { t: homePageT } = useTranslation('seo-home-page');
  if (props.page === 'user')
    return {
      title: userPageT('title'),
      image:
        'https://assets.website-files.com/613a4aeb1cd38a810f6a1448/61689732f4b0536181967320_Sem-Ti%CC%81tulo-1.png',
      description: userPageT('description'),
      url: `https://isaac.com.br/minhaconta`,
      ogType: 'website',
    };
  return {
    title: homePageT('title'),
    image:
      'https://assets.website-files.com/613a4aeb1cd38a810f6a1448/61689732f4b0536181967320_Sem-Ti%CC%81tulo-1.png',
    description: homePageT('description'),
    url: 'https://isaac.com.br',
    ogType: 'website',
  };
};

export default useSeo;
