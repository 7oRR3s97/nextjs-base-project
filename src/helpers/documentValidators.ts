import range from 'lodash/range';

export const CPFValidator = (data: string) => {
  const cpf = data.replace(/\.|-/g, '');

  if (
    cpf.length !== 11 ||
    cpf === '00000000000' ||
    cpf === '11111111111' ||
    cpf === '22222222222' ||
    cpf === '33333333333' ||
    cpf === '44444444444' ||
    cpf === '55555555555' ||
    cpf === '66666666666' ||
    cpf === '77777777777' ||
    cpf === '88888888888' ||
    cpf === '99999999999'
  )
    return false;

  // Validate first digit
  const firstDigitSum = range(0, 9).reduce((sum, item) => {
    sum += parseInt(cpf.charAt(item), 10) * (10 - item);
    return sum;
  }, 0);

  const firstDigitRemainder = 11 - (firstDigitSum % 11);
  const firstDigit = [10, 11].includes(firstDigitRemainder)
    ? 0
    : firstDigitRemainder;
  if (firstDigit !== parseInt(cpf.charAt(9), 10)) return false;

  // Validate second digit
  const secondDigitSum = range(0, 10).reduce((sum, item) => {
    sum += parseInt(cpf.charAt(item), 10) * (11 - item);
    return sum;
  }, 0);

  const secondDigitRemainder = 11 - (secondDigitSum % 11);
  const secondDigit = [10, 11].includes(secondDigitRemainder)
    ? 0
    : secondDigitRemainder;
  if (secondDigit !== parseInt(cpf.charAt(10), 10)) return false;

  return true;
};

export const CNPJValidator = (data: string) => {
  const cnpj = data.replace(/\.|\/|-/g, '');
  const size = cnpj.length - 2;
  const digits = cnpj.substring(size);
  const firstNumbers = cnpj.substring(0, size);

  if (
    cnpj.length !== 14 ||
    cnpj === '00000000000000' ||
    cnpj === '11111111111111' ||
    cnpj === '22222222222222' ||
    cnpj === '33333333333333' ||
    cnpj === '44444444444444' ||
    cnpj === '55555555555555' ||
    cnpj === '66666666666666' ||
    cnpj === '77777777777777' ||
    cnpj === '88888888888888' ||
    cnpj === '99999999999999'
  )
    return false;

  // Valida DVs
  const { sum: firstSum } = range(size, 0, -1).reduce(
    ({ sum, position }, item) => {
      sum += parseInt(firstNumbers.charAt(size - item), 10) * position;
      position -= 1;
      if (position < 2) position = 9;
      return { sum, position };
    },
    { sum: 0, position: size - 7 }
  );

  const firstResult = firstSum % 11 < 2 ? 0 : 11 - (firstSum % 11);
  if (firstResult !== parseInt(digits.charAt(0), 10)) return false;

  const currentSize = size + 1;
  const currentNumbers = cnpj.substring(0, currentSize);

  const { sum: secondSum } = range(currentSize, 0, -1).reduce(
    ({ sum, position }, item) => {
      sum += parseInt(currentNumbers.charAt(currentSize - item), 10) * position;
      position -= 1;
      if (position < 2) position = 9;
      return { sum, position };
    },
    { sum: 0, position: currentSize - 7 }
  );

  const secondResult = secondSum % 11 < 2 ? 0 : 11 - (secondSum % 11);
  if (secondResult !== parseInt(digits.charAt(1), 10)) return false;

  return true;
};

export const NIFValidator = (documentData: string) => {
  if (
    documentData.length < 9 ||
    (documentData.substring(0, 1) !== '1' && // pessoa singular
      documentData.substring(0, 1) !== '2' && // pessoa singular
      documentData.substring(0, 1) !== '3' && // pessoa singular
      documentData.substring(0, 2) !== '45' && // pessoa singular não residente
      documentData.substring(0, 1) !== '5' && // pessoa colectiva
      documentData.substring(0, 1) !== '6' && // administração pública
      documentData.substring(0, 2) !== '70' && // herança indivisa
      documentData.substring(0, 2) !== '71' && // pessoa colectiva não residente
      documentData.substring(0, 2) !== '72' && // fundos de investimento
      documentData.substring(0, 2) !== '77' && // atribuição oficiosa
      documentData.substring(0, 2) !== '79' && // regime excepcional
      documentData.substring(0, 1) !== '8' && // empresário em nome individual (extinto)
      documentData.substring(0, 2) !== '90' && // condominios e sociedades irregulares
      documentData.substring(0, 2) !== '91' && // condominios e sociedades irregulares
      documentData.substring(0, 2) !== '98' && // não residentes
      documentData.substring(0, 2) !== '99') // sociedades civis
  )
    return false;

  const nifWithoutLastDigit = documentData.slice(0, documentData.length - 1);

  const remainder =
    range(9, 1, -1).reduce((sum, item) => {
      sum += parseInt(documentData.charAt(9 - item), 10) * item;
      return sum;
    }, 0) % 11;

  if ([0, 1].includes(remainder))
    return (documentData === `${nifWithoutLastDigit}0`)!!;
  return (documentData === `${nifWithoutLastDigit}${11 - remainder}`)!!;
};

export const documentValidator = (
  documentData: string,
  documentType: string,
  country: string
) => {
  if (country.toLowerCase() === 'portugal') return NIFValidator(documentData);
  if (country.toLowerCase() === 'brasil') {
    if (documentType.toLowerCase() === 'cpf') return CPFValidator(documentData);
    if (documentType.toLowerCase() === 'cnpj')
      return CNPJValidator(documentData);
    return false;
  }
  return false;
};
