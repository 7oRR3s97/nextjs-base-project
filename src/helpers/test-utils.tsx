import React from 'react';
import {
  render as defaultRender,
  renderHook as defaultRendesHook,
  RenderHookOptions,
  RenderHookResult,
  act,
  RenderOptions,
} from '@testing-library/react';
import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import { I18nextProvider } from 'react-i18next';
import { ThemeProvider } from '@mui/material/styles';
import { Provider } from 'react-redux';

import { store } from 'redux/store';

import theme from 'styles/theme';
import QueryClientProvider from 'components/Providers/queryClientProvider';

const ns = [
  'account-data',
  'auth-alerts',
  'common',
  'edit-data',
  'home-page-callout',
  'login',
  'menu',
  'seo-home-page',
  'seo-user-page',
];
const supportedLngs = ['pt-br', 'en'];

i18n.use(initReactI18next).init({
  fallbackLng: 'en',
  debug: false,
  interpolation: {
    escapeValue: false,
  },
  defaultNS: 'common',
  ns,
  supportedLngs,
  react: {
    useSuspense: false,
  },
  resources: {
    en: {},
    'pt-br': {},
    pt: {},
  },
});

const Wrapper: React.FC = ({ children }) => {
  return (
    <Provider store={store}>
      <QueryClientProvider>
        <I18nextProvider i18n={i18n}>
          <ThemeProvider theme={theme}>{children}</ThemeProvider>
        </I18nextProvider>
      </QueryClientProvider>
    </Provider>
  );
};

const render = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper' | 'store'>
) => {
  return defaultRender(ui, { wrapper: Wrapper, ...options });
};

const renderHook = <Result, Props>(
  render: (initialProps: Props) => Result,
  options?: Omit<RenderHookOptions<Props>, 'wrapper'>
): RenderHookResult<Result, Props> => {
  return defaultRendesHook(render, { wrapper: Wrapper, ...options });
};

const advanceTimersByTime = async (time: number) =>
  await act(async () => {
    jest.advanceTimersByTime(time);
  });

/* istanbul ignore next */
export * from '@testing-library/react';
export { render, renderHook, advanceTimersByTime };
