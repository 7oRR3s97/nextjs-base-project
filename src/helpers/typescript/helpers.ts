export const objectKeys = <TObj>(obj: TObj): (keyof TObj)[] => {
  return Object.keys(obj) as (keyof TObj)[];
};

export const getDynamicValueFromObject = <TObj>(
  obj: TObj,
  index: string
): TObj[keyof TObj] | undefined => {
  if (!Object.keys(obj).includes(index)) return;
  return obj[index as keyof TObj];
};
