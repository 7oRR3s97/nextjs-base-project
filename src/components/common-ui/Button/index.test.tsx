import React from 'react';
import { render, screen, cleanup } from 'helpers/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import { composeStories } from '@storybook/testing-react';
import Button from './index';

import * as ButtonStories from './index.stories';

expect.extend(toHaveNoViolations);

describe('Button', () => {
  afterEach(() => {
    cleanup();
  });

  const stories = Object.values(composeStories(ButtonStories)).map((Story) => {
    return {
      name: Story.storyName!,
      Story,
    };
  });

  it.each(stories)(
    'should have no accessibility error in $name button story',
    async ({ Story }) => {
      const { container } = render(<Story />);
      const AxeResults = await axe(container);
      expect(AxeResults).toHaveNoViolations();
    }
  );

  it.each(stories)('should render a $name button from story', ({ Story }) => {
    render(<Story data-testid="button-component" />);

    const button = screen.queryByTestId('button-component');
    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render a Filled Button', async () => {
    render(
      <Button
        data-testid="button-component"
        layout="filled"
        fullWidth={false}
        isLoading={false}
        disabled={false}
        newWindow={false}
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render a loading filled button', () => {
    render(
      <Button
        data-testid="button-component"
        layout="filled"
        fullWidth={false}
        isLoading
        disabled={false}
        newWindow={false}
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render an Icon Filled button', () => {
    render(
      <Button
        data-testid="button-component"
        layout="filled"
        fullWidth={false}
        isLoading={false}
        icon={<div />}
        disabled={false}
        newWindow={false}
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render a href Filled Button', () => {
    render(
      <Button
        data-testid="button-component"
        layout="filled"
        fullWidth={false}
        isLoading={false}
        href="https://kizi.com.br"
        disabled={false}
        newWindow={false}
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render a href newWindow Filled Button', () => {
    render(
      <Button
        data-testid="button-component"
        layout="filled"
        fullWidth={false}
        isLoading={false}
        href="https://kizi.com.br"
        newWindow
        disabled={false}
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });

  it('should render a disabled Filled Button', () => {
    render(
      <Button
        data-testid="button-component"
        fullWidth={false}
        isLoading={false}
        disabled
      >
        button
      </Button>
    );

    const button = screen.queryByTestId('button-component');

    expect(button).toMatchSnapshot();
    expect(button).toBeInTheDocument();
  });
});
