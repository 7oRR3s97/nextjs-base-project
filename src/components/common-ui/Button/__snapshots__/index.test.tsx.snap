// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render a Alert button from story 1`] = `
<button
  class="MuiButton-root MuiButton-alert MuiButton-alertPrimary MuiButton-sizeMedium MuiButton-alertSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1lysfs6-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Box button from story 1`] = `
<button
  class="MuiButton-root MuiButton-box MuiButton-boxPrimary MuiButton-sizeMedium MuiButton-boxSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-11gj9jf-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Filled Button 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Filled button from story 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a FilledBlack button from story 1`] = `
<button
  class="MuiButton-root MuiButton-filledBlack MuiButton-filledBlackPrimary MuiButton-sizeMedium MuiButton-filledBlackSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-16zyl76-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a FilledBoxShadowMain button from story 1`] = `
<button
  class="MuiButton-root MuiButton-filledBoxShadowMain MuiButton-filledBoxShadowMainPrimary MuiButton-sizeMedium MuiButton-filledBoxShadowMainSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-bxeksc-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a FilledBoxShadowWhite button from story 1`] = `
<button
  class="MuiButton-root MuiButton-filledBoxShadowWhite MuiButton-filledBoxShadowWhitePrimary MuiButton-sizeMedium MuiButton-filledBoxShadowWhiteSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-eu9je8-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a FilledWhite button from story 1`] = `
<button
  class="MuiButton-root MuiButton-filledWhite MuiButton-filledWhitePrimary MuiButton-sizeMedium MuiButton-filledWhiteSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-ah7gmo-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Footer button from story 1`] = `
<button
  class="MuiButton-root MuiButton-footer MuiButton-footerPrimary MuiButton-sizeMedium MuiButton-footerSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-17brtdj-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Gradient button from story 1`] = `
<button
  class="MuiButton-root MuiButton-gradient MuiButton-gradientPrimary MuiButton-sizeMedium MuiButton-gradientSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-giz3fw-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Gray button from story 1`] = `
<button
  class="MuiButton-root MuiButton-gray MuiButton-grayPrimary MuiButton-sizeMedium MuiButton-graySizeMedium MuiButton-disableElevation MuiButtonBase-root  css-n0b1s1-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a GrayOutlined button from story 1`] = `
<button
  class="MuiButton-root MuiButton-grayOutlined MuiButton-grayOutlinedPrimary MuiButton-sizeMedium MuiButton-grayOutlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1bak55n-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Halfline button from story 1`] = `
<button
  class="MuiButton-root MuiButton-halfline MuiButton-halflinePrimary MuiButton-sizeMedium MuiButton-halflineSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-15rxd3t-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Icon button from story 1`] = `
<button
  class="MuiButton-root MuiButton-icon MuiButton-iconPrimary MuiButton-sizeMedium MuiButton-iconSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-12yegwm-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a IconUnderlined button from story 1`] = `
<button
  class="MuiButton-root MuiButton-iconUnderlined MuiButton-iconUnderlinedPrimary MuiButton-sizeMedium MuiButton-iconUnderlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-p80tga-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Link button from story 1`] = `
<button
  class="MuiButton-root MuiButton-link MuiButton-linkPrimary MuiButton-sizeMedium MuiButton-linkSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-14eylvo-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Outlined button from story 1`] = `
<button
  class="MuiButton-root MuiButton-link MuiButton-linkPrimary MuiButton-sizeMedium MuiButton-linkSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-14eylvo-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Outlinedblack button from story 1`] = `
<button
  class="MuiButton-root MuiButton-outlinedBlack MuiButton-outlinedBlackPrimary MuiButton-sizeMedium MuiButton-outlinedBlackSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-13c2xbh-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Quiz button from story 1`] = `
<button
  class="MuiButton-root MuiButton-quiz MuiButton-quizPrimary MuiButton-sizeMedium MuiButton-quizSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-vzaalj-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a SelectedUnderlined button from story 1`] = `
<button
  class="MuiButton-root MuiButton-selectedUnderlined MuiButton-selectedUnderlinedPrimary MuiButton-sizeMedium MuiButton-selectedUnderlinedSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-ppnppl-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a SquareUpDown button from story 1`] = `
<button
  class="MuiButton-root MuiButton-squareUpDown MuiButton-squareUpDownPrimary MuiButton-sizeMedium MuiButton-squareUpDownSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-g15kb8-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a Text button from story 1`] = `
<button
  class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1ksbsgc-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a disabled Filled Button 1`] = `
<button
  class="MuiButton-root MuiButton-link MuiButton-linkPrimary MuiButton-sizeMedium MuiButton-linkSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled  css-14eylvo-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  disabled=""
  tabindex="-1"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a href Filled Button 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a href newWindow Filled Button 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render a loading filled button 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root Mui-disabled  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  disabled=""
  tabindex="-1"
  type="button"
>
  <div
    class="MuiBox-root css-10ymegu"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
      role="progressbar"
      style="width: 1rem; height: 1rem;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;

exports[`Button should render an Icon Filled button 1`] = `
<button
  class="MuiButton-root MuiButton-filled MuiButton-filledPrimary MuiButton-sizeMedium MuiButton-filledSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-1hdtpvp-MuiButtonBase-root-MuiButton-root"
  data-testid="button-component"
  tabindex="0"
  type="button"
>
  <div
    class="MuiBox-root css-w4ynmh"
  >
    <div />
  </div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-116ddpc-MuiTypography-root"
  >
    button
  </p>
</button>
`;
